services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java

```diff
@@ -0,0 +1,274 @@
+
+import static java.lang.String.format;
+
+import org.openkilda.messaging.Destination;
+import org.openkilda.messaging.Message;
+import org.openkilda.messaging.Utils;
+import org.openkilda.messaging.command.BatchCommandsRequest;
+import org.openkilda.messaging.command.CommandData;
+import org.openkilda.messaging.command.CommandGroup;
+import org.openkilda.messaging.command.CommandMessage;
+import org.openkilda.messaging.command.flow.SwapFlowEndpointRequest;
+import org.openkilda.messaging.error.CacheException;
+import org.openkilda.messaging.error.ClientErrorMessage;
+import org.openkilda.messaging.error.ErrorData;
+import org.openkilda.messaging.error.ErrorMessage;
+import org.openkilda.messaging.error.ErrorType;
+import org.openkilda.messaging.error.MessageException;
+import org.openkilda.messaging.info.InfoMessage;
+import org.openkilda.messaging.info.flow.FlowResponse;
+import org.openkilda.messaging.info.flow.SwapFlowResponse;
+import org.openkilda.messaging.model.FlowDto;
+import org.openkilda.model.Cookie;
+import org.openkilda.model.FlowPair;
+import org.openkilda.model.UnidirectionalFlow;
+import org.openkilda.pce.AvailableNetworkFactory;
+import org.openkilda.pce.PathComputerConfig;
+import org.openkilda.pce.PathComputerFactory;
+import org.openkilda.pce.exception.UnroutableFlowException;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.wfm.error.ClientException;
+import org.openkilda.wfm.error.FeatureTogglesNotEnabledException;
+import org.openkilda.wfm.error.FlowNotFoundException;
+import org.openkilda.wfm.share.flow.resources.FlowResourcesConfig;
+import org.openkilda.wfm.share.flow.resources.FlowResourcesManager;
+import org.openkilda.wfm.share.mappers.FlowMapper;
+import org.openkilda.wfm.topology.AbstractTopology;
+import org.openkilda.wfm.topology.flow.FlowTopology;
+import org.openkilda.wfm.topology.flow.StreamType;
+import org.openkilda.wfm.topology.flow.service.FeatureToggle;
+import org.openkilda.wfm.topology.flow.service.FeatureTogglesService;
+import org.openkilda.wfm.topology.flow.service.FlowCommandFactory;
+import org.openkilda.wfm.topology.flow.service.FlowCommandSender;
+import org.openkilda.wfm.topology.flow.service.FlowService;
+import org.openkilda.wfm.topology.flow.validation.FlowValidationException;
+import org.openkilda.wfm.topology.flow.validation.FlowValidator;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.NonNull;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.topology.base.BaseRichBolt;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+public class FlowOperationsBolt extends BaseRichBolt {
```

Comment:
Shoudn't it be implemented following hub&spoke approach?

Reply:
Could you create a task to re-implement swap operation in flowhs topology?

Reply:
New issue created https://github.com/telstra/open-kilda/issues/2449